source/common/router/router.cc

```diff
@@ -534,8 +534,10 @@ void Filter::onResponseTimeout() {
 void Filter::onUpstreamReset(UpstreamResetType type,
                              const absl::optional<Http::StreamResetReason>& reset_reason) {
   ASSERT(type == UpstreamResetType::GlobalTimeout || upstream_request_);
+  const std::string reset_reason_str =
```

Comment:
Can we do this inline below so we don't allocate a string unless we need to?

Reply:
I was trying to optimize the case where it is logged and body is also updated so that it is created once - but if you think inlining is better, will inline at both places